packages/core/src/enqueue_links/shared.ts

```diff
@@ -160,15 +160,17 @@ export function filterRequestsByPatterns(requests: Request[], patterns?: UrlPatt
 
     const filtered: Request[] = [];
 
-    for (const urlPatternObject of patterns) {
-        const { regexp, glob } = urlPatternObject;
+    for (const request of requests) {
+        for (const urlPatternObject of patterns) {
+            const { regexp, glob } = urlPatternObject;
 
-        for (const request of requests) {
             if (
                 (regexp && request.url.match(regexp)) || // eslint-disable-line
                 (glob && minimatch(request.url, glob, { nocase: true }))
             ) {
                 filtered.push(request);
+                // Break the pattern loop, as we already matched this request once
```

Comment:
![Code - 2022-08-30 at 14 02 07@2x](https://user-images.githubusercontent.com/17960496/187420496-fe8673ee-0409-4fce-906a-a8d040ce0de9.png)
Doesn't seem like it to me

Reply:
I am not following your test, it does not print anything inside the loop? have you tried mine? It will break the whole loop. 

<img width="244" alt="image" src="https://user-images.githubusercontent.com/615580/187422211-42aacf80-9983-495c-8db0-826699f20432.png">

Reply:
Doesn't seem to be the case here 🤔, am I missing something? Do I need to use..._labels_ 🥶 
![Code - 2022-08-30 at 14 13 47@2x](https://user-images.githubusercontent.com/17960496/187422692-2aaa532a-d5bf-4e49-b038-f656a9f45c33.png)

Reply:
> Also your test only has 1 iteration anyways, so of course it will break the whole loop

Uh, what do you mean, it iterates array of 3 items. Yes, the place where I break does have only one item, that does not matter at all, as I break there. If you comment break in my test, you will see this:

<img width="244" alt="image" src="https://user-images.githubusercontent.com/615580/187427916-6ce67eab-f68d-4a52-a78a-15e16f80fb84.png">

Like I am not sure if this is issue with your code, but your comment says something else than how it works.

Reply:
The comment just clarifies that the breaks prevent the rest of the patterns from being tested on the current request in the iteration 😕. Am I missing something? Did I word it wrong?

Reply:
The comment says "Break the pattern loop", but breaking in JS will break the top level (requests) loop. 

As I mentioned, I havent really checked your code (so maybe it works, but for a different reason), but the comment seems wrong, as this should break the requests loop. My snippet does prove it works this way, or is it me who is missing something? :D

Reply:
Break does *not* break the top level loop (as seen from my screenshot) but the loop its in! (to break the top loop we'd need to use labels which are very niche) In your case, your break statement is making you confused because you have just one array you're iterating over. If you added another `[1, 2, 3]` to your test case you'll see it working and not breaking the whole loop

Reply:
lol, you are of course right 🤦